llvm/include/llvm/IR/Attributes.h

```diff
@@ -226,6 +236,10 @@ class Attribute {
   /// attribute to be a ConstantRange attribute.
   ConstantRange getValueAsConstantRange() const;
 
+  /// Return the attribute's value as a const range list. This requires the
+  /// attribute to be a const range list attribute.
+  SmallVector<std::pair<int64_t, int64_t>, 16> getValueAsRanges() const;
```

Comment:
Is it possible to return an ArrayRef instead of a copy? Will the underlying storage lifetime work out?